### PR TITLE
Remove .vsix from Wix installer

### LINKED ISSUE #5979: Remove VSIX from installer
**Do you want to request a *feature* or report a *bug*?**
feature

**What is the current behavior?**
The Visual Studio 2015-2017 vsix is included in the GE installer with an option to install.
The install option is copying the vsix to the GE installation dir. Installation can be done from the Program Files folder. This is not clear in the installer.
This option in the installer installed the old VS addin, the vsix was likely just added in a similar way

VSIX is officially distributed on VS Marketplace.

The vsix is a simple wrapper to call GE. It is visually updated in 3.0. In master VS2019 is enabled, that should be the official release.
 
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**
 * Remove vsix from the installer.
 * Add vsix as an artifact in Appveyor builds
 * Possibly add vsix to downloads at release builds
 * Add the vsix in master to Marketplace

**Environment you encounter the issue:**

- Git Extensions 3.00.00.4433
- Build fca7cf228b481ee8c1b779cf7b882ccdfbdcd1bc
- Git 2.20.1.windows.1
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.7.3260.0
- DPI 96dpi (no scaling)

**Did this work in previous version of GitExtensions (which)?**
no

### PR DESCRIPTION
**Pull request by gerhardol**
Upload vsix as a separate artifact to Appveyor

Fixes #5979

## Proposed changes

- Remove .vsix from Wix installer. It was included in the bundle, ticking installation only unpacked the file to the GE install dir.
- Upload vsix as a separate artifact to Appveyor

The current handling is a remain from when the old VS addin ran from the GE installer.
The vsix can be installed from the marketplace, but the current version should be offered for PR/master builds and I see no reason to not offer the vsix for releases too.

### Before

image

### After

image

## Test methodology 

- Manual iinstall
 
## Test environment(s)

### CODE REVIEW

**Comment by RussKie** [comment]
Things will need to do:
* setup a new build pipeline in Appveyor (or we could try Azure DevOps)
* setup a new signing project/definition in SignPath

I guess we could even automate publishing a new release onto the VS Market place.

**Comment by gerhardol** [comment]
I suggest to open a new issue for the suggestions and merge this as is. @RussKie ?

**Comment by RussKie** [comment]
> merge this as is

Yes, please liaise with @stefan-wenig to get a correct signing policy for the amended artifacts, and to make sure 3.0.0 builds get signed correctly as well.

File: appveyor.yml [hunk 1]
```diff
@@ -161,6 +162,7 @@ for:
           Exit-AppVeyorBuild
           return
       }
+      Push-AppveyorArtifact .\GitExtensionsVSIX\bin\Release\GitExtensionsVSIX.vsix
```
Comment (RussKie) [inline appveyor.yml]: Having a second thought, I don't think this is a correct way.
The artifact should be sent for signing (if signing is enabled and it is a release build), or it should be published unsigned otherwise.
All of that logic is in `Setup\Sign-Artifacts.ps1`.

The most up-to date version of `Setup\Sign-Artifacts.ps1` is in 3.0.2 branch, it may worth porting it over to the master (it will happen, a matter of timing).

Comment (gerhardol) [reply appveyor.yml]: This (untested as I have not made any release builds in this branch) will upload the vsix unmodified.
Signing the vsix is an enhancement as I see it, it is not done now. Should likely be done, even if VS has not complained.
Must this be done in this PR?
(I can look into this regardless, I got privileges to SignPath project.)

Comment (sharwell) [inline appveyor.yml]: ❓ Why is this line needed? It seems like a duplicate of the line above:

https://github.com/gitextensions/gitextensions/blob/da096174401e285472afd077418e9f9d584d58cb/appveyor.yml#L121

Comment (RussKie) [reply appveyor.yml]: The VSIX is signed as part of the MSI installer, you can see the rules if you log into SignPath portal.

Comment (RussKie) [reply appveyor.yml]: We have 2 distinct paths in the appveyor:
* all non-release branches: https://github.com/gitextensions/gitextensions/blob/master/appveyor.yml#L83-L90
* release branches: https://github.com/gitextensions/gitextensions/blob/master/appveyor.yml#L122-L126

And we publish different artifacts for each.
In fact we publish different artifacts for release braches depending on whether they are from an official release (i.e. signed) or not.

Comment (RussKie) [reply appveyor.yml]: Thinking out loud

v2.x - artifacts don't get signed
v3.0 - VSIX is included into MSI, it gets signed as part of it
v3.1+ - VSIX is separate, MSI signing must continue to work as previously. We may need a new signing policy for signing VSIX

Comment (gerhardol) [reply appveyor.yml]: Signing is primarily done to ensure install time integrity. The 3.0 vsix is therefore not signed at installation time. It should be OK to distribute the vsix in the 3.0 and 3.1+ way, the official distribute is in the vs marketplace anyway. But it may be required to sign the marketplace file and it should be similar for the non-marketplace vsix as it is for non signed GE installer.

I will have to read about this. But signing the vsix should not be critical.

Comment (gerhardol) [reply appveyor.yml]: It may be possible to add an "artifacts: path:" section for release too, I choose to follow the pattern for release. But it is hard to test.

**Comment by gerhardol** [comment]
I suggest this is merged as is and a new issue is created to sign  the .vsix and possibly separate the .vsix to a separate repo.

**Comment by RussKie** [comment]
👍
